airflow/dags/buildings_ingestion_gcs_dag.py

```diff
@@ -55,49 +74,82 @@ def upload_to_gcs(bucket, object_name, local_file):
     schedule_interval="@daily",
     start_date=days_ago(1),
     dagrun_timeout=timedelta(minutes=15)
-) as dag:
+):  # Because you don't use `dag` in any way, you don't have to declare it using `as`
 
-    import requests
-    from bs4 import BeautifulSoup
-    import re
+    # You know this code is guaranteed to run when this script is executed, so put these imports above with the rest of
+    # your import statements. I think it's discouraged behavior to put import statements inside of function calls,
+    # unless the imports are really big and you don't guarantee that the function is ever going to be called at all in
+    # the lifetime of the program
     url = 'https://data.openei.org/s3_viewer?bucket=oedi-data-lake&prefix=pv-rooftop/buildings/'
     reqs = requests.get(url)
     soup = BeautifulSoup(reqs.text, 'html.parser')
     urls = []
     count = 0
+
+    # I'm not familiar with BeautifulSoup but this search feels extremely generic/broad and I'm wondering if whatever
+    # you're trying to do here can be done in a better/cleaner/prettier way than just searching for the letter 'a'?
     for link in soup.find_all('a'):
-        url = 'https://data.openei.org/'+link.get('href')
+        url = 'https://data.openei.org/' + link.get('href', "")  # is it guaranteed that .get will return a string here?
+                                                                 # If not, add a default value to the call that it will
+                                                                 # fall back to
         reqs = requests.get(url)
         soup = BeautifulSoup(reqs.text, 'html.parser')
         soup = soup.find('tr', class_="odd")
+        # What's happening here? Why is this in a `while` loop? Can't you just put the `try` and `finally` here as-is?
+        # Using `while True` is really risky because you could create an infinite loop. It's generally reserved for
+        # running a program from __main__. Here, it _seems_ like you're trying to just do a single action, the action
+        # happens, and then you exit the loop, so I don't understand why you're in a loop at all.
         while True:
```

Comment:
This code should simply be:

```python
                url = soup.find('a')['href']
                urls.append(url)
```
_UNLESS_ `soup.find` has the possibility of throwing an exception, in which case:

```python
try:
    url = soup.find('a')['href']
    urls.append(url)
except (name of exception that can be thrown would be here):
    # Do something that safely handles the exception

```

No need for the `while True` at all, nor the `finally: break`